### DISCUSSION TITLE
SSG for sites with multiple hostnames (using getStaticProps)

### DISCUSSION THREAD

**Discussion by tomsoderlund** [vercel/next.js#17060] [Ideas]
I’m building a site with the structure `domain.ext/articles/articleSlug`, however it will have different content on different domains (different hostnames).

A typical use case of this would be different subdomains for different language translations of the same content: en.mysite.com, fr.mysite.com, etc.

As far as I know, this is impossible on Next.js without using `getServerSideProps` (which becomes SSR → slower performance and unnecessary if content rarely changes).

One solution could be an optional `/pages/[hostname]/...` structure.

**Update:** I thought of a workaround with using an API route `/pages/api/[hostname]/...` and then SWR to get the data but 1) [is not recommended (“You should not use fetch() to call an API route in your application”)](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) and 2) API routes are not statically generated at build time from what I can tell.

**Comment by petrbela** [comment]
In Next.js 10, localization mapping based on (sub)domain is now possible https://nextjs.org/docs/advanced-features/i18n-routing#domain-routing

However, the general use case of mapping a dynamic subdomain (such as for SaaS tenants) to be available in getStaticProps is still not supported afaik (see https://github.com/vercel/next.js/discussions/12848)

**Comment by tomsoderlund** [comment]
Here’s an example of hosting **multiple domains on the same Next.js site** (while maintaining **multiple languages** and static site generation/SSG), using Next.js’ i18n system:

https://github.com/tomsoderlund/nextjs-multi-domain-locale

**Reply by Skarian** [reply]
Will test this out this weekend. Thanks @tomsoderlund this looks promising. Any idea of its limits?

**Reply by tomsoderlund** [reply]
@Skarian Not yet! See the Todo in the Readme.

**Comment by abduraufsherkulov** [comment]
How could we achieve that with Next 13?
